### PR TITLE
Conversion from mysql to PDO

### PR DESCRIPTION
**Pull request by iqbalmalik89**
Conversion from mysql to PDO because mysql is deprecated.

### CODE REVIEW

File: book/from_flat_php_to_symfony2.rst [hunk 1]
```diff
@@ -264,14 +265,13 @@ an individual blog result based on a given id::
...
         $link = open_database_connection();
-
+    
         $id = intval($id);
-        $query = 'SELECT created_at, title, body FROM post WHERE id = '.$id;
-        $result = mysql_query($query);
-        $row = mysql_fetch_assoc($result);
-
+    	$result = $link->query('SELECT created_at, title, body FROM post WHERE id = '.$id);    
+    	$row = $result->fetch(PDO::FETCH_ASSOC);
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: this seems like a wrong indentation

File: book/from_flat_php_to_symfony2.rst [hunk 2]
```diff
@@ -264,14 +265,13 @@ an individual blog result based on a given id::
     function get_post_by_id($id)
     {
         $link = open_database_connection();
-
+
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: please do not introduce trailing whitespace

File: book/from_flat_php_to_symfony2.rst [hunk 3]
```diff
@@ -150,28 +151,28 @@ of the application are isolated in a new file called ``model.php``:
     // model.php
     function open_database_connection()
     {
-        $link = mysql_connect('localhost', 'myuser', 'mypassword');
-        mysql_select_db('blog_db', $link);
-
+    	$link = new PDO("mysql:host=localhost;dbname=blog_db", 'myuser', 'mypassword');
```
Comment (WouterJ) [inline book/from_flat_php_to_symfony2.rst]: indentation should be fixed here as well

**Comment by WouterJ** [comment]
I like updating this to not use mysql. Can you please fix the whitespace issues?

File: book/from_flat_php_to_symfony2.rst (hunk 1 above)
Comment (iqbalmalik89) [inline book/from_flat_php_to_symfony2.rst]: Yes I will fix all these corrections today. Thanks for checking.

**Comment by iqbalmalik89** [comment]
Yes, I am working on it.

**Comment by WouterJ** [comment]
As you can see in the diff of this PR, there are still some whitespace issues.

Anyway, we can also fix these while merging the PR, so I'm :+1: for this PR.

**Comment by iqbalmalik89** [comment]
Alright. It's my first PR. So what's the next step?

**Comment by WouterJ** [comment]
@iqbalmalik89 now this PR can be considered finished, the doc team (@weaverryan, @xabbuh, @javiereguiluz and me) can vote on the PR. After at least 2 team members voted +1 (you have mine already), the PR will be merged.

So unfortunately for you, it'll be a waiting game now.

File: book/from_flat_php_to_symfony2.rst [hunk 4]
```diff
@@ -264,11 +265,9 @@ an individual blog result based on a given id::
     function get_post_by_id($id)
     {
         $link = open_database_connection();
-
         $id = intval($id);
-        $query = 'SELECT created_at, title, body FROM post WHERE id = '.$id;
-        $result = mysql_query($query);
-        $row = mysql_fetch_assoc($result);
+      	$result = $link->query('SELECT created_at, title, body FROM post WHERE id = '.$id);
```
Comment (javiereguiluz) [inline book/from_flat_php_to_symfony2.rst]: This line looks wrongly indented. It should contain one additional white space. Same for the below line.

**Comment by javiereguiluz** [comment]
:+1: this change is in fact an urgent necessity now that `mysql` is deprecated/dead.

@iqbalmalik89 thanks for taking care of this and for this nice pull request.

**Comment by WouterJ** [comment]
Turns out you didn't have to wait too long :)

I've merged this PR into the 2.3 version of the docs. I've fixed the indentation (seems like you used some tabs) and done a some other minor fixes which are not related to this PR in 9960f9c.

Thank you for creating this great PR, @iqbalmalik89! :christmas_tree:

File: book/from_flat_php_to_symfony2.rst [hunk 5]
```diff
@@ -53,9 +53,9 @@ persisted to the database. Writing in flat PHP is quick and dirty:
             </ul>
         </body>
     </html>
-
+    
     <?php
-    mysql_close($link);
+    $link = null;
```
Comment (stof) [inline book/from_flat_php_to_symfony2.rst]: you should close the cursor here to be equivalent

**Pull request closed without merging**